pluginlib/CMakeLists.txt

```diff
@@ -22,7 +22,7 @@ add_library(${PROJECT_NAME} INTERFACE)
 target_include_directories(${PROJECT_NAME} INTERFACE
   "$<INSTALL_INTERFACE:include>")
 ament_target_dependencies(${PROJECT_NAME} INTERFACE
-  ament_index_cpp class_loader rcutils rcpputils tinyxml2_vendor TinyXML2)
+  ament_index_cpp class_loader rcutils rcpputils TinyXML2)
```

Comment:
The TinyXML2 interface target is only defined by the vendor package (see ros2/tinyxml2_vendor#8). So I fail to see how this can work in all the cases when the vendor package is not find_packaged before.

Reply:
`ament_target_dependencies` has nothing to do with `find_package`, does it? It ultimately calls `target_link_libraries`, isn't that right?

If so, this change makes sense as there is no target `tinyxml2_vendor::tinyxml2_vendor` or the equivalent `tinyxml2_vendor_LIBRARIES` which we could link against.
And I think there should not be one. The target to link against is `TinyXML2` and the vendor package only provides the `TinyXML2` and its respective `FindTinyXML2.cmake` file, but should not itself define a target.
From what I understood, the line below, `ament_export_dependencies(.. tinyxml2_vendor)` makes sure that a call to `find_package(tinyxml2_vendor)` will always happen. 

But I might oversee something here.

Reply:
You are right. Let me move my comment to other changes line.

Reply:
Never mind. This change in combination with the next line exporting both `tinyxml2_vendor` and `TinyXML2` makes the manual logic in the extra file obsolete.